### PR TITLE
Gaussian Troubleshooting Cleanup

### PR DESCRIPTION
**Pull request by calvinp0**
Cleaned up a couple of issue with the Gaussian Troubleshooting and added some additional tests for the Troubleshooting
Also added new trsh methods

### CODE REVIEW

File: arc/job/adapters/gaussian.py [hunk 1]
```diff
@@ -267,17 +267,25 @@ def write_input_file(self) -> None:
             self.level.method = 'cbs-qb3'
 
         # Job type specific options
-        max_c = self.args['trsh'].split()[1] if 'max_cycles' in self.args['trsh'] else 100
+        max_c = 100
+        if 'trsh' in self.args and 'trsh' in self.args['trsh']:
+            for item in self.args['trsh']['trsh']:
```
Comment (alongd) [inline arc/job/adapters/gaussian.py:272]: Are you sure we use `trsh` twice as a key?

File: arc/job/adapters/gaussian.py [hunk 2]
```diff
@@ -267,17 +267,25 @@ def write_input_file(self) -> None:
...
+                match = re.search(r'maxcycle=(\d+)', item)
+                if match:
+                    max_c = int(match.group(1))
+                    break
+
```
Comment (alongd) [inline arc/job/adapters/gaussian.py:277]: can you remove one of these empty lines?

File: arc/job/adapters/gaussian.py [hunk 3]
```diff
@@ -267,17 +267,25 @@ def write_input_file(self) -> None:
...
+
         if self.job_type in ['opt', 'conformers', 'optfreq', 'composite']:
-            keywords = ['ts', 'calcfc', 'noeigentest', f'maxcycles={max_c}'] if self.is_ts else ['calcfc']
+            keywords = ['ts', 'calcfc', 'noeigentest', f'maxcycle={max_c}'] if self.is_ts else ['calcfc']
             if self.level.method in ['rocbs-qb3']:
                 # There are no analytical 2nd derivatives (FC) for this method.
-                keywords = ['ts', 'noeigentest', 'maxcycles=100'] if self.is_ts else []
+                keywords = ['ts', 'noeigentest', f'maxcycle={max_c}'] if self.is_ts else []
```
Comment (alongd) [inline arc/job/adapters/gaussian.py:283]: [here ](https://gaussian.com/opt/)(under Options) it says `MaxCycles`, please check

**Review by alongd — COMMENTED** [review]
Thanks! I added a few comments

File: arc/job/adapters/gaussian.py (hunk 3 above)
Comment (calvinp0) [reply arc/job/adapters/gaussian.py:283]: @alongd 
maxcycle/maxcyc/maxcycles are all accepted parameters of Gaussian

Comment (calvinp0) [reply arc/job/adapters/gaussian.py:283]: https://github.com/ReactionMechanismGenerator/ARC/pull/758#discussion_r1712952653

File: arc/job/adapters/gaussian.py (hunk 1 above)
Comment (calvinp0) [reply arc/job/adapters/gaussian.py:272]: Yes

**Review by alongd — APPROVED** [review]
Thanks!

**Pull request merged**